prometheus/rootfs/bin/boot.sh

```diff
@@ -22,18 +22,23 @@ done
 
 set +e
 
-OPTS=""
+until [ -n "${DEIS_MONITOR_ALERT_SERVICE_HOST+1}" ]; do
```

Comment:
Another option would be to, by convention, use a set of hostnames to wire this stuff up. So static configuration would always be `http://deis-monitor-alertmanager:80` for the current namespace, or `http://deis-monitor-alertmanager.deis-system.svc:80` for reaching into a separate namespace (or, whatever namespace we use for the Deis control plane).

Then the boot logic can loop until that hostname resolves or the container can exit and be restarted by the kubelet.

To allow operators to override that configuration, these hostnames could then be overridden by use of specific ENV variables in the pod templates rather than using the k8s downward API.